src/core/qgsgdalutils.cpp

```diff
@@ -90,6 +97,65 @@ gdal::dataset_unique_ptr QgsGdalUtils::createSingleBandTiffDataset( QString file
   return hDstDS;
 }
 
+gdal::dataset_unique_ptr QgsGdalUtils::imageToMemoryDataset( const QImage &image, QgsRectangle extent, const QgsCoordinateReferenceSystem &crs )
+{
+  if ( image.isNull() )
+    return nullptr;
+  QByteArray red, green, blue, alpha;
+
+  const QRgb *rgb = reinterpret_cast<const QRgb *>( image.constBits() );
+
+  // potentially an overflow here -- that's on Qt!
```

Comment:
is this comment true ? I guess that on 32 bit builds, Qt won't let you allocate an image with more than 4 billion pixels